schematic/utils/general.py

```diff
@@ -26,3 +27,30 @@ def unlist(_list):
         return _list[0]
     else:
         return _list
+
+def get_dir_size(path='.'):
+    '''
+    calculate total size of a directory
+    args: 
+    path: path to a folder or directory
+    '''
+    total = 0
+    with os.scandir(path) as it:
```

Comment:
The most common usecase of context managers are for patching/mocking and opening files in Python.  Due to this, I set out to find out why a context manager should be used here and this is what I found: https://github.com/python/cpython/issues/70182

Nice!

Reply:
I double checked by looking at the documentation related to `os.scandir` [here](https://docs.python.org/3/library/os.html#os.scandir). It seems like that even though `scandir.close()` will be called automatically, it is still advisable to use the with statement. I have left a comment about this in the code.